vignettes/applications.Rmd

```diff
@@ -0,0 +1,249 @@
+
+library(data.table)
+end.i <- fit$changepoints
+start.i <- c(1, end.i[-length(end.i)]+1)
+segments.dt <- with(ECG$data, data.table(
+  timeStart=time[start.i],
+  timeEnd=time[end.i],
+  state=fit$states,
+  mean=fit$means))
+
+segments.dt[, letter := c("beforeQ", "Q", "R", "S", "S1", "S2", "peak", "afterPeak", "foo")[state+1] ]
+mean.dt <- segments.dt[, data.table(
+  time=as.numeric(rbind(timeStart-0.5, timeEnd+0.5)),
+  mean=as.numeric(rbind(mean, mean)))]
+
+m <- function(x){
+  factor(x, c("previous", "changepoint"), c(
+    "Previous model", "Proposed model"))
+}
+model.dt <- rbind(
+  segments.dt[, data.table(
+    model=m("changepoint"),
+    time=ifelse(letter=="Q", timeEnd, (timeStart+timeEnd)/2),
+    millivolts=mean,
+    letter)],
+  data.table(
+    model=m("previous"),
+    ECG$PanTompkins)
+)[letter %in% c("Q", "R", "S")]
+
+samples.per.second <- 250
+truth.dt <- segments.dt[letter=="R", list(time=(timeStart+timeEnd)/2)]
+library(ggplot2)
+gg <- ggplot()+
+  geom_vline(aes(
+    xintercept=time/samples.per.second),
+    color="red",
+    data=truth.dt)+
+  geom_text(aes(
+    x, y, hjust=hjust, label="True R"),
+    color="red",
+    size=3,
+    data=data.table(
+      x=208.5, y=6500, hjust=1, label="True R", model=m("changepoint")))+
+  theme_bw()+
+  theme(panel.margin=grid::unit(0, "lines"))+
+  facet_grid(model ~ .)+
+  geom_line(aes(
+    time/samples.per.second, millivolts),
+    color="grey50",
+    data=ECG$data)+
+  geom_line(aes(
+    time/samples.per.second, mean),
+    data=data.table(model=m("changepoint"), mean.dt),
+    color="blue")+
+  geom_label(aes(
+    time/samples.per.second, millivolts,
+    label=letter),
+    color="blue",
+    size=3,
+    label.padding=grid::unit(0.1, "lines"),
+    alpha=0.6,
+    data=model.dt)+
+  coord_cartesian(xlim=c(52000, 52900)/samples.per.second, expand=FALSE)+
+  xlab("Time (seconds)")+
+  ylab("Electrocardiogram activity (mV)")
+##png("figure-ecg.png", 7, 2.5, res=300, units="in")
+print(gg)
+##dev.off()
+
+```
+
+
+## Multi-modal regression for neuro spike train data set 
+
+
+```{r, neuro, fig.width=10}
+
+library(data.table)
+data(neuroSpike, package="gfpop")
+fps <- 100
+seconds.between.data <- 1/fps
+sec.w <- seconds.between.data/3
+myGraph <- gfpop::graph(
+  gfpop::edge(0, 0, "down", 0),
+  gfpop::edge(1, 1, "up",   0),
+  gfpop::edge(0, 1, "up",   5),
+  gfpop::edge(1, 0, "down", 0),
+  gfpop::edge(0:1, 0:1, "null"))
+
+fit <- gfpop::gfpop(
+  vectData = neuroSpike$calcium, mygraph = myGraph, type = "gauss")
+
+end.i <- fit$changepoints
+start.i <- c(1, end.i[-length(end.i)]+1)
+res.dt <- with(neuroSpike, data.table(
+  start.seconds=seconds[start.i],
+  end.seconds=seconds[end.i],
+  state=fit$states,
+  mean=fit$means))
+res.dt[, Multimodal := ifelse(mean<0.1, 0, mean) ]
+over.dt <- res.dt[neuroSpike, on=list(
+  start.seconds <= seconds, end.seconds >= seconds)]
+tall.dt <- melt(
+  over.dt,
+  measure.vars=c("Multimodal", "AR1"),
```

Comment:
right now I am using a pre-computed AR1 model.

Is the exp-decay operator implemented? If so it would be better to show how the AR1 model can be computed via gfpop.